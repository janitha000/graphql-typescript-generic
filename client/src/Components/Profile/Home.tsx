import React from 'react'
import About from './About'
import Contact from './Contact'
import Navbar from './Navbar'
import './Profile.css'

const Home: React.FC = () => {
    return (
        <div className="profile">
            <Navbar />
            <div className="profile-sections">
                <About />
                <Contact />
            </div>

        </div>
    )
}

export default Home
