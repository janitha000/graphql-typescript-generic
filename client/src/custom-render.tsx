import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router"
import { UserContextProvider } from "./Contexts/UserContext"
import React from 'react'



const Wrapper = ({ children }: any) => {
    return (
        <UserContextProvider>
            <MemoryRouter>
                {children}
            </MemoryRouter>
        </UserContextProvider>
    )
}

const customRender = (ui: any, options?: any) =>
    render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react'

export { customRender as render }