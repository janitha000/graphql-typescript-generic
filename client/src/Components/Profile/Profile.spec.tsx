import React from 'react'
import { render } from '../../custom-render'
import Home from './Home'
import renderer from 'react-test-renderer'
import { cleanup } from '@testing-library/react'

describe('Profile component', () => {
    afterEach(() => {
        cleanup()
    })

    it('should render component', () => {
        const { getByText } = render(<Home />)
        expect(getByText('This is about')).toBeInTheDocument()
    })

    it('should match the snapshots', () => {
        const tree = renderer.create(<Home />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})