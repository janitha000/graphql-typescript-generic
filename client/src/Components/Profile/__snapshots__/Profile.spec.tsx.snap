// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should match the snapshots 1`] = `
<div
  className="profile"
>
  <div
    className="profile__navbar"
  >
    This is nav bar
  </div>
  <div
    className="profile-sections"
  >
    <div
      className="profile__about"
    >
      This is about
    </div>
    <div
      className="profile__contact"
    >
      This is contact page
    </div>
  </div>
</div>
`;
